Avoid closing datastore before all dependent services are stopped

#### Version information:

> 0.5.0-dev

#### Description:

Currently, can get errors like the following because we stop the datastore before we're done with it. We need to synchronously shut down the provider queue.

```
ipfs cat zCT5htkdztJi3x4zBNHo8TRvGHPLTdHUdCLKgTGMgQcRKSLoWxK1
Cid4Data2020-03-13T14:40:10.986+0100    ERROR   provider.queue  queue/queue.go:93       error querying for head of queue: datastore closed, stopping provider
github.com/ipfs/go-ipfs-provider/queue.(*Queue).work.func1
        pkg/mod/github.com/ipfs/go-ipfs-provider@v0.4.1/queue/queue.go:93
```

--- a/go.mod
+++ b/go.mod
@@ -38,7 +38,7 @@ require (
 	github.com/ipfs/go-ipfs-files v0.0.8
 	github.com/ipfs/go-ipfs-pinner v0.0.4
 	github.com/ipfs/go-ipfs-posinfo v0.0.1
-	github.com/ipfs/go-ipfs-provider v0.4.2
+	github.com/ipfs/go-ipfs-provider v0.4.3
 	github.com/ipfs/go-ipfs-routing v0.1.0
 	github.com/ipfs/go-ipfs-util v0.0.1
 	github.com/ipfs/go-ipld-cbor v0.0.4

